File: tracks/typescript/exercises/05-human-in-the-loop/practice/worker.ts
import { Worker } from '@temporalio/worker';

async function main(): Promise<void> {
  const worker = await Worker.create({
    workflowsPath: require.resolve('./workflows'),
    taskQueue: 'approval-tasks',
  });
  console.log("Worker started on task queue 'approval-tasks'. Ctrl-C to stop.");
  await worker.run();
}

main().catch((error: unknown) => {
  console.error(error);
  process.exit(1);
});

File: tracks/python/exercises/05-human-in-the-loop/practice/worker.py
import asyncio

from temporalio.client import Client
from temporalio.worker import Worker

from workflows import CampaignApprovalWorkflow


async def main():
    client = await Client.connect("localhost:7233")
    worker = Worker(
        client,
        task_queue="approval-tasks",
        workflows=[CampaignApprovalWorkflow],
    )
    print("Worker started on task queue 'approval-tasks'. Ctrl-C to stop.")
    await worker.run()


if __name__ == "__main__":
    try:
        asyncio.run(main())
    except KeyboardInterrupt:
        print("Worker stopped.")
